app/workers/search/chat_channel_membership_es_sync_worker.rb

```diff
@@ -0,0 +1,17 @@
+module Search
+  class ChatChannelMembershipEsSyncWorker
```

Comment:
It may get repetitive to define this over and over again, so I made you a module:

```ruby
module EsIndexWorker
  extend ActiveSupport::Concern

  included do
    include Sidekiq::Worker

    sidekiq_options queue: :high_priority

    def perform(id)
      klass = self.class.name.demodulize.chomp("EsSyncWorker")
      object = ObjectSpace.const_get(klass).find_by(id: id)

      if object
        object.index_to_elasticsearch_inline
      else
        Search.const_get(klass).delete_document(id)
      end
    end
  end
end
```

Then this and similar workers would become 

```ruby
module Search
  class ChatChannelMembershipEsSyncWorker
    include EsIndexWorker
  end
end
```

WDYT? I only did some cursory testing in a Rails console, but it seems to be doing what it should.

Reply:
Rather than do a module I was actually thinking even simpler for the next refactor and make a single Sync job like this
```ruby
module Search
  class ElasticsearchSyncWorker
    include Sidekiq::Worker

    sidekiq_options queue: :high_priority

    VALID_RECORD_TYPES = %w[ChatChannelMembership ClassifiedListing Tag].freeze

    def perform(record_type, id)
      unless VALID_RECORD_TYPES.include?(record_type)
        raise InvalidRecordType, "Invalid class: #{record_type}. Valid and indexable classes are #{VALID_RECORD_TYPES.join(', ')}"
      end

      record = "::#{record_type}".safe_constantize.find_by(id: id)

      if record
        record.index_to_elasticsearch_inline
      else
        "Search::#{record_type}".safe_constantize.delete_document(id)
      end
    end
  end
end
```

Reply:
@mstruve I wasn't sure if you wanted to keep them separate in case they ever need to diverge, but if that's not the case, a single job is definitely easier.

Reply:
I think we may diverge some of them in the future to handle things like batch indexing but I do think many will run through this flow for a while at the very least and it may always be a good idea to have a single job for single document indexing